src/diffusers/models/upsampling.py

```diff
@@ -165,6 +165,14 @@ def forward(self, hidden_states: torch.Tensor, output_size: Optional[int] = None
         # if `output_size` is passed we force the interpolation output
         # size and do not make use of `scale_factor=2`
         if self.interpolate:
+            # upsample_nearest_nhwc also fails when the number of output elements is large
+            # https://github.com/pytorch/pytorch/issues/141831
+            scale_factor = (
+                2 if output_size is None else max([f / s for f, s in zip(output_size, hidden_states.shape[-2:])])
```

Comment:
`[f / s for f, s in zip(output_size, hidden_states.shape[-2:])]`

-> this could be unpacked into a variable with a sensible name and reused?

Reply:
I'm not sure, adding a variable that is only used when `output_size` is provided (which is not the usual case) can be more confusing, in my opinion, and I don't think we should complicate things further with an `if` branch. Also the value will never be reused. Perhaps I can add another line to the comment instead.

Reply:
I don't see how assigning `max([f / s for f, s in zip(output_size, hidden_states.shape[-2:])])` to a variable is not a better thing to do here as far as readability is concerned. But I am not going sweat over it, either. So, up to you.

Reply:
Readability is important to me, I'm happy to adapt to any recommendations! To be more specific, these are the options I see:

This PR:
```
        if self.interpolate:
            scale_factor = (
                2 if output_size is None else max([f / s for f, s in zip(output_size, hidden_states.shape[-2:])])
            )
```

With a var:
```
        if self.interpolate:
             if output_size is not None:
                 max_scale_factor = max([f / s for f, s in zip(output_size, hidden_states.shape[-2:])]
            scale_factor = 2 if output_size is None else max_scale_factor
```

Or:
```
        if self.interpolate:
             if output_size is None:
                  scale_factor = 2
              else:
                 scale_factor = max([f / s for f, s in zip(output_size, hidden_states.shape[-2:])]
```

I don't like the long expression in this PR, but I find it more difficult to understand the flow at a glance in the second alternative with the branches. The third one looks less Pythonic to me.

Happy to use what the team prefers :)

Reply:
Thanks for the consideration. I would prefer the third option. We don't go strictly Pythonic some times to have more explicit nature. But will let others comment,